### DISCUSSION TITLE
More MariaDB errors when trying to start afresh with PhotoPrism

### DISCUSSION THREAD

**Discussion by captainkev76** [photoprism/photoprism#1680] [Q&A] [answered]
Hi all,

I had some problems with permissions for Maria DB last week, which I think I've now sorted out, however I'm now getting a bunch of errors which seem to indicate a corrupt database when I'm trying to set up a fresh installation of Photoprism on a fresh Ubuntu 20.04 server (in Virtualbox).

The errors are as follows:
```
mariadb_1     | 2021-10-28 15:41:36+00:00 [Note] [Entrypoint]: Entrypoint script for MariaDB Server 1:10.6.4+maria~focal started.
mariadb_1     | 2021-10-28 15:41:36+00:00 [Note] [Entrypoint]: Initializing database files
mariadb_1     | 2021-10-28 15:41:36 0 [ERROR] InnoDB: Corrupted page [page id: space=0, page number=0] of datafile './ibdata1' could not be found in the doublewrite buffer.
mariadb_1     | 2021-10-28 15:41:36 0 [ERROR] InnoDB: Plugin initialization aborted with error Data structure corruption
mariadb_1     | 2021-10-28 15:41:36 0 [ERROR] Plugin 'InnoDB' init function returned error.
mariadb_1     | 2021-10-28 15:41:36 0 [ERROR] Plugin 'InnoDB' registration as a STORAGE ENGINE failed.
mariadb_1     | 2021-10-28 15:41:36 0 [ERROR] Unknown/unsupported storage engine: InnoDB
mariadb_1     | 2021-10-28 15:41:36 0 [ERROR] Aborting

```
Has anyone come across this before?

[EDITED TO SHOW CORRECT ERROR LOGS]

**Comment by graciousgrey** [comment]
Did you go through the checklists we provide --> https://docs.photoprism.org/getting-started/troubleshooting/

**Reply by captainkev76** [reply]
Yes, I went through this. Very useful info but alas didn't resolve the issue

**Comment by lastzero** [comment]
Did you upgrade / change the database version e.g. run MariaDB 10.5 instead of 10.6?

**Reply by lastzero** [reply]
see https://docs.photoprism.org/getting-started/troubleshooting/#upgrading-mariadb

**Reply by captainkev76** [reply]
Yes, I've tried 10.5 and 10.6 and they both have the same issue.

**Comment by lastzero** [comment, marked as answer]
Looking at MariaDB docs for your error message, it seems your InnoDB tables need repair:
- https://mariadb.com/kb/en/repairing-mariadb-tables-for-sql-server-users/#innodb

You can add the `--innodb-force-recovery` parameter just like you would add the `--skip-grant-tables` parameter in this example:

https://docs.photoprism.org/getting-started/troubleshooting/#lost-root-password

Note there are multiple recovery modes which you can set via `--innodb-force-recovery=MODE`:
- https://mariadb.com/kb/en/innodb-recovery-modes/

Docs say you should try `--innodb-force-recovery=0` before exploring other options.

**Reply by captainkev76** [reply]
Yeah, been down those avenues too. The problem manifests even when it's setting up a fresh instance, when there are no existing tables to be corrupted. It's definitely unique to the situation where the database is located on a shared network drive, as it sets up perfectly when non-persistant storage is used. I raised a ticket on the MariaDB Github page and they've asked me to log it as a bug, which I've done. Initial feedback is that the log entries generated by the error are too vague to allow the exact problem to be isolated. Personally, I'm not 100% sure it's a bug and not something I've done (or not done) to mount the network drives correctly and set permissions correctly. I've checked everything I can but I know from experience that when I think I've found a bug, it's normally me that's the bug :-)

**Reply by lastzero** [reply]
Never use a shared network drive for storing database files. It's the exact problem you would expect from this, for example because of caching or sync issues.

**Reply by lastzero** [reply]
Sorry, recovery flag should have been `--innodb-force-recovery=1`

Updated our docs and docker-compose.yml examples accordingly:
- https://docs.photoprism.org/getting-started/troubleshooting/#corrupted-files
- https://docs.photoprism.org/getting-started/docker-compose/#database
- https://dl.photoprism.org/docker/docker-compose.yml

**Comment by afrojewelz** [comment]
same issues came back ,by adding those params could not solve this issue in my case. except my db is on my local ssd drive and fully writeable
